core/capabilities/syncer.go

```diff
@@ -98,36 +104,42 @@ func (s *registrySyncer) Start(ctx context.Context) error {
 	}
 	myId := s.peerWrapper.GetPeer().ID().String()
 	config := remotetypes.RemoteTriggerConfig{
-		RegistrationRefreshMs: 20000,
+		RegistrationRefreshMs:   20000,
+		MinResponsesToAggregate: 2,
 	}
 	if slices.Contains(workflowDONPeers, myId) {
 		s.lggr.Info("member of a workflow DON - starting remote subscribers")
-		triggerCap := remote.NewTriggerSubscriber(config, triggerInfo, capabilityDonInfo, workflowDonInfo, s.dispatcher, nil, s.lggr)
+		aggregator := triggers.NewMercuryRemoteAggregator(s.lggr)
+		triggerCap := remote.NewTriggerSubscriber(config, triggerInfo, triggerCapabilityDonInfo, workflowDonInfo, s.dispatcher, aggregator, s.lggr)
 		err = s.registry.Add(ctx, triggerCap)
 		if err != nil {
 			s.lggr.Errorw("failed to add remote target capability to registry", "error", err)
 			return err
 		}
-		err = s.dispatcher.SetReceiver(capId, capabilityDonInfo.ID, triggerCap)
+		err = s.dispatcher.SetReceiver(capId, triggerCapabilityDonInfo.ID, triggerCap)
 		if err != nil {
-			s.lggr.Errorw("failed to set receiver", "capabilityId", capId, "donId", capabilityDonInfo.ID, "error", err)
+			s.lggr.Errorw("failed to set receiver", "capabilityId", capId, "donId", triggerCapabilityDonInfo.ID, "error", err)
 			return err
 		}
 		s.subServices = append(s.subServices, triggerCap)
 	}
-	if slices.Contains(capabilityDONPeers, myId) {
+	if slices.Contains(triggerDONPeers, myId) {
 		s.lggr.Info("member of a capability DON - starting remote publishers")
 		workflowDONs := map[string]remotetypes.DON{
 			workflowDonInfo.ID: workflowDonInfo,
 		}
-		underlying := &noOpTrigger{info: triggerInfo, lggr: s.lggr}
-		triggerCap := remote.NewTriggerPublisher(config, underlying, triggerInfo, capabilityDonInfo, workflowDONs, s.dispatcher, s.lggr)
-		err = s.dispatcher.SetReceiver(capId, capabilityDonInfo.ID, triggerCap)
+		underlying := triggers.NewMercuryTriggerService(1000, s.lggr)
+		triggerCap := remote.NewTriggerPublisher(config, underlying, triggerInfo, triggerCapabilityDonInfo, workflowDONs, s.dispatcher, s.lggr)
+		err = s.dispatcher.SetReceiver(capId, triggerCapabilityDonInfo.ID, triggerCap)
 		if err != nil {
-			s.lggr.Errorw("failed to set receiver", "capabilityId", capId, "donId", capabilityDonInfo.ID, "error", err)
+			s.lggr.Errorw("failed to set receiver", "capabilityId", capId, "donId", triggerCapabilityDonInfo.ID, "error", err)
 			return err
```

Comment:
maybe out of scope in this change, but i'd like to see these error be wrapped with some useful msg ~ fmt.Errorf("failed to set reciever: %w", err)

without the error wrapping the reported error in higher in the call stack is kind of useless; you must have the log and not the error message in order to have any hope of debugging.

given the pattern here, maybe we want a helper that wraps the logger and both does the structured logging and returns a properly wrapped error

Reply:
Valid suggestion but I think I'll skip it here because this Syncer implementation is temporary and will be replaced by one reading from onchain registry.